pkg/agent/route/route_windows.go

```diff
@@ -329,11 +320,97 @@ func (c *Client) UnMigrateRoutesFromGw(route *net.IPNet, linkName string) error
 }
 
-// Run is not supported on Windows and returns immediately.
+// Run periodically syncs netNatStaticMapping and route. It will not return until stopCh is closed.
 func (c *Client) Run(stopCh <-chan struct{}) {
+	klog.InfoS("Starting netNatStaticMapping and route sync", "interval", SyncInterval)
+	wait.Until(c.syncIPInfra, SyncInterval, stopCh)
 }
 
-func (c *Client) listRoutes() (map[string]*util.Route, error) {
+// syncIPInfra is idempotent and can be safely called on every sync operation.
+func (c *Client) syncIPInfra() {
+	if err := c.syncRoute(); err != nil {
+		klog.ErrorS(err, "Failed to sync route")
+	}
+
+	if c.proxyAll {
+		if err := c.syncNetNatStaticMapping(); err != nil {
+			klog.ErrorS(err, "Failed to sync netNatStaticMapping")
+		}
+	}
+	klog.V(3).Info("Successfully synced netNatStaticMapping and route")
+}
+
+func (c *Client) syncRoute() error {
+	restoreRoute := func(route *util.Route) bool {
+		if err := util.ReplaceNetRoute(route); err != nil {
+			klog.ErrorS(err, "Failed to sync route", "Route", route)
+			return false
+		}
+		return true
+	}
+	c.nodeRoutes.Range(func(_, v interface{}) bool {
+		route := v.(*util.Route)
+		return restoreRoute(route)
+	})
+	if c.proxyAll {
+		c.serviceRoutes.Range(func(_, v interface{}) bool {
+			route := v.(*util.Route)
+			return restoreRoute(route)
+		})
+	}
+	// These routes are installed automatically by the kernel when the address is configured on
+	// the interface. If these routes are deleted manually by mistake, we restore them.
+	gwAutoconfRoutes := []*util.Route{
+		{
+			LinkIndex:         c.nodeConfig.GatewayConfig.LinkIndex,
+			DestinationSubnet: util.NewIPNet(c.nodeConfig.GatewayConfig.IPv4),
+			GatewayAddress:    c.nodeConfig.GatewayConfig.IPv4,
+			RouteMetric:       util.MetricDefault,
+		},
+		{
+			LinkIndex:         c.nodeConfig.GatewayConfig.LinkIndex,
+			DestinationSubnet: c.nodeConfig.PodIPv4CIDR,
+			GatewayAddress:    c.nodeConfig.GatewayConfig.IPv4,
+			RouteMetric:       util.MetricDefault,
+		},
+	}
+
+	for _, route := range gwAutoconfRoutes {
+		restoreRoute(route)
+	}
+	return nil
+}
+
+func (c *Client) syncNetNatStaticMapping() error {
+	if err := util.NewNetNat(antreaNatNodePort, virtualNodePortDNATIPv4Net); err != nil {
+		return err
+	}
+
+	restoreNetNatStaticMapping := func(m *util.NetNatStaticMapping) bool {
+		if err := util.ReplaceNetNatStaticMapping(m); err != nil {
+			klog.ErrorS(err, "Failed to add netNatStaticMapping", "netNatStaticMapping", m)
+			return false
+		}
+		return true
+	}
+
+	c.netNatStaticMappings.Range(func(_, v interface{}) bool {
+		mapping := v.(*util.NetNatStaticMapping)
+		return restoreNetNatStaticMapping(mapping)
+	})
+
+	return nil
+}
+
+func (c *Client) isServiceRoute(route *util.Route) bool {
+	// If the gateway IP is the virtual Service IP , then it is a route which is added by AntreaProxy.
+	if route.GatewayAddress.Equal(config.VirtualServiceIPv4) || route.DestinationSubnet.IP.Equal(config.VirtualServiceIPv4) {
```

Comment:
I still feel it is needed to check route.GatewayAddress!=nil before comparing the values. I used to hit similar failures when comparing routes in VM Agent scearios.

Reply:
Done